docs/website/blog/2024-03-07-openapi-generation-chargebee.md

```diff
@@ -0,0 +1,144 @@
+---
+slug: openapi-generation-chargebee
+title: "Saving 75% of work for a Chargebee Custom Source via pipeline code generation with dlt"
+image:  https://storage.googleapis.com/dlt-blog-images/openapi-generation.png
+authors:
+  name: Adrian Brudaru & Violetta Mishechkina
+  title: Data Engineer & ML Engineer
+  url: https://github.com/dlt-hub/dlt
+  image_url: https://avatars.githubusercontent.com/u/89419010?s=48&v=4
+tags: [data observability, data pipeline observability]
+---
+At dltHub, we're pioneering the future of data pipeline generation, making complex processes simple and scalable.
+
+Pipeline generation on a simple level is already possible directly in gpt chats - just ask for it. But doing it at scale, correctly, and producing comprehensive, good quality pipelines is a much more complex endeavour.
+
+# The automated work by our existing PoC
+
+![OpenAPI spec can be used to generate multiple things](https://storage.googleapis.com/dlt-blog-images/openapi-generation.png)
+*This diagram illustrates how OpenAPI spec is used for software generation.*
+
+To generate pipelines, what we need is the right knowledge of what should happen. One such source is the OpenApi spec, but it does not contain complete information for generating everything.
+
+Last year we experimented with extracting this information and generating pipelines, and you can read about it here: https://dlthub.com/docs/blog/open-api-spec-for-dlt-init
+
+Our PoC leveraged the OpenAPI spec to automate pipeline creation, achieving significant strides in efficiency and accuracy.
+
+
+# The manual last mile
+
+Ideally, we would love to point a tool at an api or doc of the api, and just have the pipeline generated. We can already do that to some extent, and fill in any gaps manually.
+
+While automation handles the bulk, some customisation remains manual, generating requirements towards our ongoing efforts of full automation.
+
+# Why now? Community growth requires faster onboarding with code generation
+
+dlt community has been growing steadily in recent months. In February, we had a 25% growth on slack and even more in usage.
+
+The new users generated a lot of questions and some of them used our onboarding program, where we speed-run you through any obstacles, learning how to make things smoother along the way.
```

Comment:
Let's put some link to onboarding program. Can be google form https://docs.google.com/forms/d/1IsdqT5aape1xOL0Qf9AtND-lvFwv1O5qOGlJY4Q4t4M/edit. Or

```
(to join our onboarding program, request a [call](link to call)))
```